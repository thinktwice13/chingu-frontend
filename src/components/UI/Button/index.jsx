/* eslint-disable  react/button-has-type */

import React from 'react';
import PropTypes from 'prop-types';

const { node, oneOf } = PropTypes;

const Button = props => {
  const { children, ...restProps } = props;
  return <button {...restProps}>{children}</button>;
};

Button.propTypes = {
  children: node.isRequired,
  type: oneOf(['button', 'submit', 'reset']),
};

Button.defaultProps = {
  type: 'button',
};

export default Button;
