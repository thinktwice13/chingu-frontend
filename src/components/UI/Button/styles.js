import { css } from 'styled-components';
import themes from 'components/UI/styles/themes';

const sizes = {
  default: css`
    font-size: 14px;
    min-width: 150px;
  `,
  small: css`
    font-size: 10px;
    min-width: 80px;
  `,
  large: css`
    font-size: 18px;
    min-width: 200px;
  `,
};

/**
 * @description gets button size bnased on size prop
 * @param {Object} props passed to Button component
 * @param {string} props.size must be default, small or large
 *
 * @returns {Object} css for button size properties
 */
export const getSize = props => sizes[props.size];

/**
 * @description Invert button styles
 * @param {Object} s - styles
 * @param {string} s.color - primary color
 * @param {string} s.background - background color
 * @param {string} s.border - border color
 * @param {string} s.hover - hover background color
 *
 * @returns {Object} inverted styles object
 */
const invert = s => ({
  color: s.background,
  background: s.color,
  border: s.background,
  hover: '#00000011',
});

/**
 * @description gets button color styles
 * @param {Object} props - cbutton component props must have  theme, disbaled and inverted props
 */
export const getColors = props => {
  const { theme, inverted, disabled } = props;

  const t = disabled ? 'disabled' : theme;

  const { fg, bg } = themes[t];
  let styles = {
    color: fg,
    background: bg,
    border: '#00000000',
    hover: `${bg}aa`,
  };

  if (inverted) {
    styles = invert(styles);
  }

  return css`
    color: ${styles.color};
    background-color: ${styles.background};
    border-color: ${styles.border};
    :hover {
      background-color: ${disabled ? '' : styles.hover};
    }
  `;
};
