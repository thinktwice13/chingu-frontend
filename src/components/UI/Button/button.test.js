import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json'; // Needed to exclude enzyme wrapper from saved snapshots
import Button from '.';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
  it('renders with correct props', () => {
    const btn = shallow(<Button>Click me</Button>);
    expect(btn.text()).toEqual('Click me');

    btn.setProps({ type: 'submit' });
    expect(toJSON(btn)).toMatchSnapshot();
  });
});

// describe('Button', () => {
//   it('render button with correct props', () => {
//     const btn = shallow(<Button>Click me</Button>);
//     expect(btn).toMatchSnapshot();

//     btn.setProps({ disabled: true });
//     expect(btn.render()).toMatchSnapshot();

//     btn.setProps({ inverted: true });
//     expect(btn.render()).toMatchSnapshot();

//     btn.setProps({ size: 'small' });
//     expect(btn.render()).toMatchSnapshot();

//     btn.setProps({ size: 'large' });
//     expect(btn.render()).toMatchSnapshot();

//     btn.setProps({ rounded: true });
//     expect(btn.render()).toMatchSnapshot();

//     btn.setProps({ style: { backgroundColor: 'orange' } });
//     expect(btn.render()).toMatchSnapshot();
//   });

//   it('renders as anchor tag with href prop', () => {
//     const btn = shallow(<Button href='http://www.chingu.io'>Click me</Button>);
//     expect(btn.find('a').exists()).toBe(true);
//     expect(btn.find('button').exists()).toBe(false);
//   });

//   it('calls event listeners', () => {
//     const click = jest.fn();

//     const btn = shallow(<Button onClick={click}>Click me</Button>);

//     btn.simulate('click');
//     expect(click).toBeCalled();

//     btn.simulate('click');
//     btn.simulate('click');
//     expect(click).toBeCalledTimes(3);
//   });
// });
