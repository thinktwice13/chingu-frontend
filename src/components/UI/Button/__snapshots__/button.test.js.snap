// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with correct props 1`] = `
<button
  type="submit"
>
  Click me
</button>
`;
